sdk/messaging/azservicebus/receiver_test.go

```diff
@@ -554,8 +555,10 @@ func TestReceiver_RenewMessageLock(t *testing.T) {
 	endCaptureFn := test.CaptureLogsForTest()
 	defer endCaptureFn()
 	expectedLockBadError := receiver.RenewMessageLock(context.Background(), messages[0], nil)
-	// String matching can go away once we fix #15644
-	// For now it at least provides the user with good context that something is incorrect about their lock token.
+
+	var asSBError Error
+	require.ErrorAs(t, expectedLockBadError, &asSBError)
```

Comment:
Presumably this succeeds at present due to the `Unwrap()` implementation on `Error`?